packages/driver/src/primitives/buffer.ts

```diff
@@ -19,28 +19,49 @@
 import type char from "./chars";
 import * as chars from "./chars";
 import { LegacyHeaderCodes } from "../ifaces";
-import { Buffer } from "../adapter.node";
-
-/* WriteBuffer over-allocation */
-const BUFFER_INC_SIZE = 4096;
-
-const EMPTY_BUFFER = new Uint8Array(0);
 
 export const utf8Encoder = new TextEncoder();
 export const utf8Decoder = new TextDecoder("utf8");
 
-const decodeB64 = (b64: string): Uint8Array => {
-  return Buffer.from(b64, "base64");
-};
-const encodeB64 = (data: Uint8Array): string => {
-  const buf = Buffer.isBuffer(data)
-    ? data
-    : Buffer.from(data.buffer, data.byteOffset, data.byteLength);
-  return buf.toString("base64");
-};
+let decodeB64: (_: string) => Uint8Array;
+let encodeB64: (_: Uint8Array) => string;
+
+try {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  const Buffer = require("node:buffer").Buffer;
+
+  decodeB64 = (b64: string) => {
+    return Buffer.from(b64, "base64");
+  };
+  encodeB64 = (data: Uint8Array) => {
+    const buf = Buffer.isBuffer(data)
+      ? data
+      : Buffer.from(data.buffer, data.byteOffset, data.byteLength);
+    return buf.toString("base64");
+  };
+} catch (_) {
+  decodeB64 = function (b64: string) {
+    const binaryString = atob(b64);
```

Comment:
BTW, I think this isn't unicode-safe, see [this note on MDN](https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem). Their implementation of a unicode safe codec is VERY slow, so one thing we could do would be to use the `base64-js` package that `buffer` uses, which is almost as fast as `Buffer`.

We should probably also add a test case for encoding and decoding unicode buffers.

Reply:
@scotttrinh but isn't here the string always base64 encoded string? so it is actually safe... 
I believe that decodeB64 means: decode base64 encoded string? and encodeB64: encode to base64 string.

Reply:
Oh, sorry, I think the issue is on the `encodeB64` side: If the bytes are beyond `0xFF`, it'll throw an error at runtime. Maybe this isn't a problem in practice, but we should maybe add a test using the same test string in the linked MDN article?

Reply:
How can a byte be greater than `0xFF`?

Reply:
Yep, you're right, not the bytes, but the code points of the input string.

> Since btoa interprets the code points of its input string as byte values, calling btoa on a string will cause a "Character Out Of Range" exception if a character's code point exceeds 0xff. 

The fact that we are dealing with `Uint8Array`s already means I think we're safe here since the encoding itself will take care of ensuring that the binary data is split into utf-8 bytes correctly. Just did some quick in-browser testing and it _does_ look like this works as expected. I'd still like a test to make sure we don't break it in the future, but we can add that test separately.